app/utilities/nick.py

```diff
@@ -0,0 +1,24 @@
+import re
+from typing import NamedTuple
+
+REGEX_EXTRACT_NAME_AND_NUMBER = r"^([\[『](?P<num>\d+)[\]』])?\s*(?P<name>.*?)\s*([\[『](?P<num2>\d+)[\]』])?$"
+
+
+class LabMember(NamedTuple):
+    name: str | None
+    lab_member_number: int | None
+
+
+def extract_lab_member_number(name: str) -> "LabMember":
+    """Split the username into name and lab member number (if available).
+    """
+
+    match = re.search(REGEX_EXTRACT_NAME_AND_NUMBER, name)
+
+    name = match.group("name")
+    number = match.group("num") or match.group("num2")
+
+    return LabMember(
+        name=name,
+        lab_member_number=int(number) if number else None
+    )
```

Comment:
What is going to be the use case for the LabMember class? We can fetch their lab member number from the database easily already, so creating an object to pass this information around feels a bit excessive, but maybe there's more info I'm not seeing. Maybe you could share more info on this, or someone else can comment on this. (This also goes for test file).

Reply:
But what is the use case for this? Why would we need this instead of just getting the data via the database? The separator function is a nice touch to have though.

Reply:
I actually thought about it a little more and realize it's utility. Resolving.